### DISCUSSION TITLE
Alexa compatibility

### DISCUSSION THREAD

**Discussion by dp247** [robbrad/UKBinCollectionData#1325] [General]
I haven't used Amazon's Alexa platform in years, but could probably maybe build a skill that feeds off this.

**Comment by robbrad** [comment]
@dp247 - I have started looking into this

**Comment by robbrad** [comment]
Still on my radar - just sorting some of the repo hygiene

**Comment by robbrad** [comment]
@dp247 / @OliverCullimore 

Started looking at this and I think we are going to need some compute like an AWS Lambda (this acts as an endpoint to serve the content). Now I know some of the councils take a long time to run so I think we need a new Github action to run all councils daily and store the data in an S3 bucket. The Lambda would then use the Alexa invocation to get the "next bin data" from the s3 bucket

**Comment by dp247** [comment]
Not to get ahead of myself, but could the S3 bucket also help some of the other ideas (like #446 and #458)?

**Comment by robbrad** [comment]
It's a difficult one to implement in an efficient manner - as each household is going to need a its own data, which from the experience of the councils can take anywhere from 2 to 60 seconds per pull - this will drive up infrastructure costs.

**Comment by robbrad** [comment]
Been thinking about this one and im just not sure its going to work as a live intergration.

What if you used our custom component to add Bin Data and then used Alexa with a Text-To-Speach approach from Home Assistant ?

eg: 
```
lounge_temp:
  alias: Lounge temp
  sequence:
  - service: notify.lounge_dot
    data_template:
      message: "The temperature is {{states('sensor.broadlink_sensor_temperature')|replace('.', ' point ') }} degrees"
```

**Comment by raldred** [comment]
I currently do this with to broadcast to google home speakers. using the google assistant SDK and a couple of custom helper entities, an automation triggers the evening before the next bin day, and announces the bins being collected tomorrow.

I don't think there's any need to host any data.
Maybe a simple blueprint where people can configure the entities to trigger on?

That being said, I don't know if a similar broadcast system exists for Alexa in home assistant.

**Comment by robbrad** [comment]
@dp247 Finally done it - and it only took me 2.5 years

**Install and Configure Alexa Media Player**

**Install via HACS (recommended):**

-  Make sure you have [HACS](https://hacs.xyz/) installed in Home Assistant.
-  Go to HACS → Integrations → Explore & Download Repositories and search for Alexa Media Player.
-  Install and restart Home Assistant.

**Configure Alexa Media Player:**

-  Go to Configuration → Integrations in Home Assistant.
-  Click Add Integration and search for Alexa Media Player.
-  Log in with your Amazon credentials.
-  After a successful login, Home Assistant will discover all your Alexa devices (Echo Dots, Echo Shows, etc.).

**Create a Home Assistant Input control**

```
input_boolean:
  bin_query_trigger:
    name: "Bin Query Trigger"
    initial: off
```

**Create an Alexa Routine**

Create an Alexa Routine in the Alexa app that starts when you say a specific phrase (e.g., "Alexa, what's my bin collection?").
In that routine, choose “Smart Home” → “Control Device” to toggle something in Home Assistant (like an input_boolean or a script).
Home Assistant sees that input_boolean or script as “turned on,” and an automation in Home Assistant fires:

The automation calls notify.alexa_media (via [Alexa Media Player](https://github.com/custom-components/alexa_media_player)) to speak the sensor data back to you.

Then Create an automation - you just need to change the sensors to match your sensor names for the _days_until_collection sensors.

You can change the Mappting name eg Garden Waste

```
alias: "Respond to Bin Query"
description: ""
trigger:
  - platform: state
    entity_id: input_boolean.bin_query_trigger
    to: "on"

action:
  # 1) Build your message once using `variables:` or `template:` approach
  - variables:
      bin_message: >-
        {% set bin_sensor_map = {
          'sensor.cheshire_east_council_empty_bin_standard_garden_waste_days_until_collection':
            'Garden Waste',
          'sensor.cheshire_east_council_empty_standard_general_waste_days_until_collection':
            'General Waste',
          'sensor.cheshire_east_council_empty_standard_mixed_recycling_days_until_collection':
            'Mixed Recycling'
        } %}
        
        {# Step 1: Find the earliest (min) days. #}
        {% set all_days = bin_sensor_map.keys() | map('states') | map('int', default=999) | list %}
        {% set min_days = all_days | min %}
        
        {# Step 2: Create a namespace to hold your final bins list. #}
        {% set bins_next = namespace(entities=[]) %}
        
        {# Step 3: Loop over the dictionary and append to bins_next.entities. #}
        {% for days_sensor, friendly_name in bin_sensor_map.items() %}
          {% if states(days_sensor) | int(default=999) == min_days %}
            {% set bins_next.entities = bins_next.entities + [friendly_name] %}
          {% endif %}
        {% endfor %}
        
        {# Step 4: Build a helper string that describes the day in words. #}
        {% if min_days == 0 %}
          {% set day_text = 'today' %}
        {% elif min_days == 1 %}
          {% set day_text = 'tomorrow' %}
        {% else %}
          {% set day_text = 'in ' ~ min_days ~ ' days' %}
        {% endif %}
        
        {# Step 5: Final output #}
        {% if min_days == 999 %}
          I can’t find any bin collection data — check your sensors.
        {% elif bins_next.entities|length == 0 %}
          I found days data, but none matched the earliest day for the bins.
        {% elif bins_next.entities|length == 1 %}
          The {{ bins_next.entities[0] }} bin is collected {{ day_text }}.
        {% else %}
          The following bins are collected {{ day_text }}:
          {{ bins_next.entities|join(' and ') }}.
        {% endif %}

  - service: alexa_media.update_last_called
    data: {}
  - delay:
      milliseconds: 500
  - service: homeassistant.update_entity
    data: {}
    target:
      entity_id: sensor.last_alexa
  - delay:
      milliseconds: 500
  # 2) Notify each Alexa device separately
  - service: notify.alexa_media_last_called
    data:
      message: "{{ bin_message }}"

  # 3) Turn off your boolean trigger
  - service: input_boolean.turn_off
    data:
      entity_id: input_boolean.bin_query_trigger

mode: single
```

**Comment by robbrad** [comment]
@m26dvd FYI
